warehouse/utils/project.py

```diff
@@ -28,15 +40,156 @@ def remove_documentation(task, request, project_name):
         task.retry(exc=exc)
 
 
-def confirm_project(project, request, fail_route):
-    confirm = request.POST.get("confirm_project_name")
+def _namespace_stdlib_list(module_list):
+    for module_name in module_list:
+        parts = module_name.split(".")
+        for i, part in enumerate(parts):
+            yield ".".join(parts[: i + 1])
+
+
+STDLIB_PROHIBITED = {
+    canonicalize_name(s.rstrip("-_.").lstrip("-_."))
+    for s in chain.from_iterable(
+        _namespace_stdlib_list(stdlib_list.stdlib_list(version))
+        for version in stdlib_list.short_versions
+    )
+}
+
+
+def add_project(name, request):
```

Comment:
One concern with this refactor is that it has side effects that I believe are incorrect for all use cases as discussed in another comment. I would suggest refactoring further to separate the concerns of "can we create this project given its name?" and "actually create the project".